app/src/main/java/clarkson/ee408/tictactoev4/LoginActivity.java

```diff
@@ -7,7 +7,21 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import android.content.Intent;
+import android.widget.Toast;
+
 import clarkson.ee408.tictactoev4.model.User;
+import clarkson.ee408.tictactoev4.client.SocketClient;
+import clarkson.ee408.tictactoev4.executor.AppExecutors;
+import clarkson.ee408.tictactoev4.network.Request;
+import clarkson.ee408.tictactoev4.network.RequestType;
+import clarkson.ee408.tictactoev4.network.Response;
+import clarkson.ee408.tictactoev4.network.ResponseStatus;
+
+import clarkson.ee408.tictactoev4.PairingActivity;
+import clarkson.ee408.tictactoev4.RegisterActivity;
```

Comment:
Some imports are coming from packages that don't exist. This section should look like this after correcting and organizing them


```suggestion
import clarkson.ee408.tictactoev4.client.AppExecutors;
import clarkson.ee408.tictactoev4.client.SocketClient;
import clarkson.ee408.tictactoev4.model.User;
import clarkson.ee408.tictactoev4.socket.Request;
import clarkson.ee408.tictactoev4.socket.RequestType;
import clarkson.ee408.tictactoev4.socket.Response;
import clarkson.ee408.tictactoev4.socket.ResponseStatus;
```